docs/tutorial/fdfitting.rst

```diff
@@ -0,0 +1,343 @@
+and fit them. We construct it using `pylake.FdFit` and passing it one or more models. In
+return, we get an object we can interact with, which in this case we store in `odijk_fit`::
+
+    odijk_fit = pylake.FdFit(odijk_with_offsets)
+
+Adding data to the fit
+----------------------
+
+Next up, is adding some data. Let's assume we have two data sets. One was acquired in the presence
+of a ligand, and another was measured without a ligand. We expect this ligand to only affect the
+contour length of our DNA. Adding the first data set to the fit is simple::
+
+    odijk_fit.add_data("Control", force1, distance1)
+
+For the second data set, we want the contour length to be different. We can achieve this by renaming
+it when loading the data::
+
+    odijk_fit.add_data("RecA", force2, distance2, params={"DNA/Lc": "DNA/Lc_RecA"})
+
+More specifically, we renamed the parameter `DNA/Lc` to `DNA/Lc_RecA`. Sometimes, you may want
+a large number of data sets with different offsets. Assuming we have two lists of distance and
+force vectors stored in the lists distances and forces. In this case, it may make sense to load
+them in a loop and set such transformations programmatically::
+
+    for i, (distance, force) in enumerate(zip(distances, forces)):
+        odijk_fit.add_data(f"RecA {i}", f=force, d=distance, params={"DNA/f_offset": f"DNA/f_offset_{i}"})
+
+The syntax `f"DNA/f_offset_{i}"` is parsed into `DNA/f_offset_0`, `DNA/f_offset_1` ... etc. For more
+information on how this works, read up on Python fantastic f-Strings.
+
+Setting parameter bounds
+------------------------
+
+The parameters of the model can be accessed directly from `FdFit`. Note that by default,
+parameters tend to have reasonable initial guesses and bounds in Pylake, but we can set
+our initial guess and a lower and upper bound as follows::
+
+    odijk_fit["DNA/Lp"].value = 50
+    odijk_fit["DNA/Lp"].lower_bound = 39
+    odijk_fit["DNA/Lp"].upper_bound = 80
+
+After this, the model is ready to be fitted::
+
+    odijk_fit.fit()
+
+Note that multiple models can be fit at once, by just supplying more than one model::
+
+    multi_model_fit = pylake.FdFit(model1, model2, model3)
+
+Frequently, global fits have better statistical properties than fitting the data separately
+as more information is available to infer parameters shared between the various models.
+
+Plotting the data
+-----------------
+
+Fits can be plotted using the built-in plot functionality::
+    
+    odijk_fit.plot()
+    plt.ylabel('Force [pN]')
+    plt.xlabel('Distance [$\\mu$M]');
+
+However, sometimes more fine grained control over the plots is required. Let's say we want to plot
+the model over the range 2.0 to 5.0 for the conditions from `Control` and `RecA`. We can do this by
+calling plot on the model directly::
+
+    odijk_with_offsets.plot(odijk_fit['Control'], np.arange(2.0, 5.0, .01), fmt='k--')
+    odijk_with_offsets.plot(odijk_fit['RecA'], np.arange(2.0, 5.0, .01), fmt='k--')
```

Comment:
You raise a good point. There should probably be an easier way for users to plot things. So... the new one:

If you have just one model: `fit.plot()` will take care of everything.

If you have more than one: `fit[model1].plot()` will plot all data corresponding to model 1. I prefer the angle brackets because it is analogous to the API elsewhere.

And you can (if you want) also plot a specific dataset: `fit.plot("data name")`. Or in the multimodel case `fit[model1].plot("data name")`

You can also pass an optional `independent` if you want to plot over a custom time range.

I think this covers all the cases that I've used in the notebooks practically, with relative ease.

I do think the model.plot will have a role for sensitivity analyses and the like, and it is called by `FdFit.plot` now too, so I would like to keep that one around too, just not make it the goto one.

Reply:
That's a great solution! I really like it!

Reply:
Cool, thanks! 👍